stuffer/s2n_stuffer_base64.c

```diff
@@ -139,61 +85,31 @@ int s2n_stuffer_write_base64(struct s2n_stuffer *stuffer, struct s2n_stuffer *in
 {
     POSIX_PRECONDITION(s2n_stuffer_validate(stuffer));
     POSIX_PRECONDITION(s2n_stuffer_validate(in));
-    s2n_stack_blob(o, 4, 4);
-    s2n_stack_blob(i, 3, 3);
-
-    while (s2n_stuffer_data_available(in) > 2) {
-        POSIX_GUARD(s2n_stuffer_read(in, &i));
-
-        /* Take the top 6-bits of the first data byte  */
-        o.data[0] = b64[(i.data[0] >> 2) & 0x3f];
-
-        /* Take the bottom 2-bits of the first data byte -  0b00110000 = 0x30
-         * and take the top 4-bits of the second data byte - 0b00001111 = 0x0f
-         */
-        o.data[1] = b64[((i.data[0] << 4) & 0x30) | ((i.data[1] >> 4) & 0x0f)];
 
-        /* Take the bottom 4-bits of the second data byte - 0b00111100 = 0x3c
-         * and take the top 2-bits of the third data byte - 0b00000011 = 0x03
-         */
-        o.data[2] = b64[((i.data[1] << 2) & 0x3c) | ((i.data[2] >> 6) & 0x03)];
+    int binary_data_size = s2n_stuffer_data_available(in);
+    if (binary_data_size == 0) {
+        return S2N_SUCCESS;
+    }
+    int base64_output_size = (binary_data_size / 3) * 4;
+    /* we will need to add a final padded block */
+    if (binary_data_size % 3 != 0) {
+        base64_output_size += 4;
+    }
+    /* Null terminator is added */
+    base64_output_size += 1;
```

Comment:
Nit: I only commented on s2n_stuffer_read_base64 before, but the same "group" variable would help readability here.
```suggestion
    int binary_data_size = s2n_stuffer_data_available(in);
    if (binary_data_size == 0) {
        return S2N_SUCCESS;
    }
    
    int base64_groups = (binary_data_size / 3)
    /* we will need to add a final padded block */
    if (binary_data_size % 3 != 0) {
        base64_groups++;
    }
    
    int base64_output_size = base64_groups * 4;
    /* Null terminator is added */
    base64_output_size += 1;
```